src/core/lifetime.d

```diff
@@ -2161,3 +2195,31 @@ pure nothrow @nogc @system unittest
     static assert(!__traits(compiles, f(ncarray)));
     f(move(ncarray));
 }
+
+// @betterC unittest
+// {
+//     static struct A
+//     {
+//         int i;
+//         this(int i)
+//         {
+//             this.i = i;
+//         }
+//         ~this()
+//         {
+//             i = 0;
+//         }
+//     }
+
+//     static struct B
+//     {
+//         A a;
+//         this(int i)
+//         {
+//             this.a = A(i);
+//         }
+//     }
+
+//     B b = void;
+//     emplace(b, 1);
+// }
```

Comment:
If you want to add this unittest, but it doesn't work yet, I suggest placing the part of it that fails under `version (none)` (that way the compiler will do at least a minimal checking of the syntax) and a FIXME comment.

Reply:
Removed.